lib/IcaoDecoder.ts

```diff
@@ -2,103 +2,202 @@ export class IcaoDecoder {
   name : string;
   icao : string;
 
+  // Pre-compiled regex patterns for military ICAO address detection (performance optimization)
+  private static readonly MILITARY_PATTERNS = {
+    // US military
+    US_MIL_1: /^adf[7-9]/,
+    US_MIL_2: /^adf[a-f]/,
+    US_MIL_3: /^a[ef]/,
+
+    // Egypt military
+    EGYPT_MIL: /^0100[78]/,
+
+    // Algeria military
+    ALGERIA_MIL: /^0a4/,
+
+    // Italy military
+    ITALY_MIL: /^33ff/,
+
+    // France military
+    FRANCE_MIL_1: /^3a[89a-f]/,
+    FRANCE_MIL_2: /^3b/,
+
+    // Germany military
+    GERMANY_MIL_1: /^3e[ab]/,
+    GERMANY_MIL_2: /^3f[4-9ab]/,
+
+    // UK military
+    UK_MIL_1: /^4000[0-3]/,
+    UK_MIL_2: /^43c/,
+
+    // Austria military
+    AUSTRIA_MIL: /^44[4-7]/,
+
+    // Belgium military
+    BELGIUM_MIL: /^44f/,
+
+    // Bulgaria military
+    BULGARIA_MIL: /^457/,
+
+    // Denmark military
+    DENMARK_MIL: /^45f4/,
+
+    // Greece military
+    GREECE_MIL: /^468[0-3]/,
+
+    // Hungary military
+    HUNGARY_MIL: /^473c0/,
+
+    // Norway military
+    NORWAY_MIL: /^4781/,
+
+    // Netherlands military
+    NETHERLANDS_MIL: /^480/,
+
+    // Poland military
+    POLAND_MIL: /^48d8[0-7]/,
+
+    // Portugal military
+    PORTUGAL_MIL: /^497c/,
+
+    // Czech Republic military
+    CZECH_MIL: /^49842/,
+
+    // Switzerland military
+    SWITZERLAND_MIL: /^4b7/,
+
+    // Turkey military
+    TURKEY_MIL: /^4b82/,
+
+    // Slovenia military
+    SLOVENIA_MIL: /^506f/,
+
+    // Oman military
+    OMAN_MIL: /^70c07/,
+
+    // Saudi Arabia military
+    SAUDI_MIL_1: /^7102[5-8]/,
+    SAUDI_MIL_2: /^7103[89]/,
+
+    // Israel military
+    ISRAEL_MIL: /^738a/,
+
+    // Australia military
+    AUSTRALIA_MIL_1: /^7c8[2-48]/,
+    AUSTRALIA_MIL_2: /^7[def]/,
+
+    // India military
+    INDIA_MIL: /^8002/,
+
+    // Canada military
+    CANADA_MIL: /^c[23]/,
+
+    // Brazil military
+    BRAZIL_MIL: /^e4[01]/,
+
+    // Chile military
+    CHILE_MIL: /^e806/,
+  };
+
   constructor(icao: string) {
     this.name = 'icao-decoder-typescript';
     this.icao = icao;
   }
 
   isMilitary() {
-    let i = this.icao;
+    const i = this.icao;
+    const p = IcaoDecoder.MILITARY_PATTERNS;
+
     return (
```

Comment:
Make `MILITARY_PATTERNS` an array of regexp instead of a dictionary and then just `return MILITARY_PATTERNS.some((p) => p.test(i));`

all the comments can move up to the definition